### PR TITLE
Add infra to collect integration kernel coverage

### PR DESCRIPTION
**Pull request by rjoleary**
The coverage is automatically collected into a tar file and saved to
/tmp/uroot-coverage whenever the kernel supports it.

Use the gcov tool to covert these gcda and gcno files to a human
readable format.

Signed-off-by: Ryan O'Leary <ryanoleary@google.com>

### CODE REVIEW

**Review by rjoleary — COMMENTED** [review]
I pulled out the changes for the tar package. This depends on #1928 + a rebase.

File: pkg/vmtest/integration.go [hunk 1]
```diff
@@ -188,6 +226,12 @@ func QEMUTest(t *testing.T, o *Options) (*qemu.VM, func()) {
 
 	return vm, func() {
 		vm.Close()
+		if !o.NoKernelCoverage {
```
Comment (GanShun) [inline pkg/vmtest/integration.go:229]: can we invert this logic somehow? ! NokernelCoverage gets confusing

**Review by GanShun — DISMISSED** [review]

File: pkg/vmtest/integration.go (hunk 1 above)
Comment (rjoleary) [reply pkg/vmtest/integration.go:229]: The default is to collect the coverage if CONFIG_GCOV is enabled, but that can be disabled by setting NoKernelCoverage.

**Comment by rjoleary** [comment]
This is ready for another review

**Review by rminnich — APPROVED** [review]

**Review by rminnich — APPROVED** [review]

**Pull request merged**